README-dev-on-docker.md

```diff
@@ -0,0 +1,114 @@
+# Developing Vespa on Docker
+
+The purpose of this document is to describe how to set up a combined development and
+run-time environment for (iteratively) developing and testing Vespa.
+
+### Start the docker container
+
+    DOCKER_IMAGE=vespaengine/vespa-dev:latest
+    SOURCE_ROOT=${HOME}/vespa/vespa
+    APP_ROOT=${HOME}/vespa/sample-apps
+
+    docker run -d -p 8080:8080 -p 8998:8998 -p 10001:10001 \
+                -v ${HOME}/.ccache:/root/.ccache:delegated \
+                -v ${HOME}/.m2:/root/.m2:delegated \
```

Comment:
Wont this mess up user's `.m2` directory on the host? Since all new files will be owned by `root`?

I did something similar when I wanted to get system tests running inside container, ended up having to replicate the host user inside the container and run everything as that user, see: https://github.com/vespa-engine/vespa/blob/64c4f16151d121414f2b3c5e44eda8f06ba033e6/docker-api/src/test/java/com/yahoo/vespa/hosted/dockerapi/RunSystemTests.java#L138:L147

Reply:
@freva Interesting, this works flawlessly for me. Is your host Mac, Windows or Linux?

Reply:
@lesters  just tried it again:
So first I downloaded the image and ran the `bootstrap` java + `mvn install` the rest inside the container: https://git.corp.yahoo.com/gist/valerijf/059d617256e5bf52f3b5d715e9209460
That worked without any problems. (By the way, `cd /source` is missing before `./bootstarap.sh java`)

Then I ran the same two commands on my host: https://git.corp.yahoo.com/gist/valerijf/d7122f2e74d03dce9db102449e29ab83
and `mvn clean` fails because it cant delete some files owned by the root in the `*/target/*` directories that were created as part of the `mvn install` on the container. 

To fix this I have to run ` sudo find -type d -name target -exec rm -rf {} \;` on host.

Reply:
Ok, so this is on Linux and the cause is the ownership of the files built within the container. On Docker for Mac, this is the same as the host user, but this is not the case on Linux.